src/planner/vectorfield/VectorFieldUtil.cpp

```diff
@@ -0,0 +1,155 @@
+  }
+  return _outputTrajectory;
+}
+
+//==============================================================================
+
+DesiredTwistFunction::DesiredTwistFunction(
+    const Twist& _twist, const Jacobian& _jacobian)
+  : dart::optimizer::Function("DesiredTwistFunction")
+  , mTwist(_twist)
+  , mJacobian(_jacobian)
+{
+}
+
+//==============================================================================
+
+double DesiredTwistFunction::eval(const Eigen::VectorXd& _qd)
+{
+  return 0.5 * (mJacobian * _qd - mTwist).squaredNorm();
+}
+
+//==============================================================================
+
+void DesiredTwistFunction::evalGradient(
+    const Eigen::VectorXd& _qd, Eigen::Map<Eigen::VectorXd> _grad)
+{
+  _grad = mJacobian.transpose() * (mJacobian * _qd - mTwist);
+}
+
+//==============================================================================
+
+bool computeJointVelocityFromTwist(
+    const Eigen::Vector6d& _desiredTwist,
+    const aikido::statespace::dart::MetaSkeletonStateSpacePtr _stateSpace,
+    const dart::dynamics::BodyNodePtr _bodyNode,
+    double _optimizationTolerance,
+    double _timestep,
+    double _padding,
+    Eigen::VectorXd* _jointVelocity)
+{
+  using dart::math::Jacobian;
+  using dart::optimizer::Problem;
+  using dart::optimizer::Solver;
+  using Eigen::VectorXd;
+  // Use LBFGS to find joint angles that won't violate the joint limits.
+  const Jacobian jacobian
+      = _stateSpace->getMetaSkeleton()->getWorldJacobian(_bodyNode);
+
+  const std::size_t numDofs = _stateSpace->getMetaSkeleton()->getNumDofs();
+  VectorXd lowerLimits(numDofs);
+  VectorXd upperLimits(numDofs);
+
+  VectorXd positions = _stateSpace->getMetaSkeleton()->getPositions();
+  VectorXd positionLowerLimits
+      = _stateSpace->getMetaSkeleton()->getPositionLowerLimits();
+  VectorXd positionUpperLimits
+      = _stateSpace->getMetaSkeleton()->getPositionUpperLimits();
+  VectorXd velocityLowerLimits
+      = _stateSpace->getMetaSkeleton()->getVelocityLowerLimits();
+  VectorXd velocityUpperLimits
+      = _stateSpace->getMetaSkeleton()->getVelocityUpperLimits();
+
+  for (std::size_t i = 0; i < numDofs; ++i)
+  {
+    const double position = positions[i];
+    const double positionLowerLimit = positionLowerLimits[i];
+    const double positionUpperLimit = positionUpperLimits[i];
+    const double velocityLowerLimit = velocityLowerLimits[i];
+    const double velocityUpperLimit = velocityUpperLimits[i];
+
+    if (position
+        < positionLowerLimit - _timestep * velocityLowerLimit + _padding)
+      lowerLimits[i] = 0;
+    else
+      lowerLimits[i] = velocityLowerLimit;
+
+    if (position
+        > positionUpperLimit - _timestep * velocityUpperLimit - _padding)
+      upperLimits[i] = 0;
+    else
+      upperLimits[i] = velocityUpperLimit;
+  }
+
+  const auto problem = std::make_shared<Problem>(numDofs);
+  problem->setLowerBounds(lowerLimits);
+  problem->setUpperBounds(upperLimits);
```

Comment:
I think we could optimize this code a bit by passing `velocityLowerLimits` and `velocityUpperLimits` instead of passing the copies of them. This could be simply done by changing the above if-statement as:
```cpp
    if (position
        < positionLowerLimit - _timestep * velocityLowerLimit + _padding)
      velocityLowerLimits[i] = 0.0;

    if (position
        > positionUpperLimit - _timestep * velocityUpperLimit - _padding)
      velocityUpperLimits[i] = 0.0;
```